packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -16,13 +17,184 @@ import type { ThinEngine } from "core/Engines/thinEngine";
 import { ToHalfFloat } from "core/Misc/textureTools";
 import type { Material } from "core/Materials/material";
 import { Scalar } from "core/Maths/math.scalar";
+import { runCoroutineSync, runCoroutineAsync, createYieldingScheduler, type Coroutine } from "core/Misc/coroutine";
 
 interface DelayedTextureUpdate {
     covA: Uint16Array;
     covB: Uint16Array;
     colors: Uint8Array;
     centers: Float32Array;
 }
+
+// @internal
+const unpackUnorm = (value: number, bits: number) => {

```

Comment:
I think everything related to data parsing/decoding should be moved to its own file (splatFileDecoder.ts?). This would help make the `GaussianSplattingMesh` class lighter and easier to maintain.

Reply:
I agree. I'll improve that part when things settle down.